File: 002-Backend/LaPlata.API/LaPlata.Domain/Models/Cartoes/Cartao.cs
using System.ComponentModel.DataAnnotations;

namespace LaPlata.Domain.Models
{
    public class Cartao : ModelBase
    {
        [Required]
        public string Nome { get; set; }
        [Required]
        public int DiaVencimento { get; set; }
        [Required]
        public int DiaFechamento { get; set; }
        public virtual List<Compra>? Compras { get; set; }
        public virtual List<Assinatura>? Assinaturas { get; set; }
        public virtual List<Fatura>? Faturas { get; set; }
    }
}

File: 003-Frontend/HomeTools.Front/src/app/interfaces/financas/Cartao.ts
import { Assinatura } from "./Assinatura";
import { Compra } from "./Compra";

export interface Cartao {
  Id: number;
  Nome: string;
  DiaVencimento: number;
  DiaFechamento: number;
  Assinaturas: Assinatura[];
  Compras: Compra[];
}
